src/content/blog/2024-12-04-differences-between-linting-and-type-checking.md

```diff
@@ -0,0 +1,248 @@
+---
+layout: post
+title: "Differences between linting and type checking"
+teaser: "Linters such as ESLint and type checkers such as TypeScript catch different areas of code defects — and are best used in conjunction with each other."
+authors:
+  - joshuakgoldberg
+categories:
+  - Storytime
+tags:
+  - Linting
+---
+
+Linters and type checkers are two kinds of tools that both analyze code and report on detected issues.
```

Comment:
I think we need a bit more in the intro. This is a bit abstract, and I think could use some grounding in reality. Something like, "If you're a JavaScript developer today, you're probably using TypeScript or ESLint, or both..."